### ISSUE TITLE
КоптеРиг фича-баг

### ISSUE THREAD

**Issue by CasperBioS** [KranX/Vangers#88]
При использовании КоптеРиг(в полете) можно на пол секунды переключится на вектор(прибор самонаведения) и одновременно выстрелить, тоесть в полете, имея ловкость пальцев получается пользоваться самонаведением.
Проблема: После переключения в воздухе спустя цикл(или некоторое время) пропадают картинки*(чернеют) содержимого трюма, а затем предметы пропадают окончательно при заходе в эскейв или телепортации в другой мир.
Иногда предметы из трюма чернели* при перевозке 2х деталей от сломаных мехосов разом.
Этот баг наблюдаю с момента выхода первого релиза в 98 году, в версии стим все так же и осталось, видимо я единственный кто этой фичей пользовался.

**Comment by stalkerg** [comment]
Да, я в курсе это проблемы. Дело в том, что там происходит очень небезопасные манипуляции со списком элементов в трюме из-за чего повреждается память. Один такой вариант я вроде починил, остальные надо смотреть, но это всё очень сложно, там целая магия с этими списками.

**Comment by CasperBioS** [comment]
Вообще как фича, эта полусекундная замена должна остаться, но для исправления бага может быть сделать запрет на кнопку замены (Включить вектор) до приземления/выкапывания.
Последнее время играю с осторожностью, стараюсь не переключать Риги до окончательного приземления\выкапывания - баг не повторялся

**Comment by Assimilator** [comment]
Добавлю:
У меня простое включение вектора заставило содержимое чернеть, при этом сейв поломался очень странно - Смог заехать в Подиш и сохраниться, а теперь при повторном заезде игра гарантированно вылетает (протестировал 5 раз).
